common/djangoapps/student/roles.py

```diff
@@ -261,6 +275,15 @@ def __init__(self, *args, **kwargs):
         super().__init__(self.ROLE, *args, **kwargs)
 
 
+@register_access_role
+class CourseLimitedStaffRole(CourseStaffRole):
+    """A Staff member of a course without access to Studio."""
+    ROLE = 'limited_staff'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
@Agrendalath, quick question: have you deliberately not passed `self.ROLE` to the constructor here? Example:
```python
@register_access_role
class CourseStaffRole(CourseRole):
    """A Staff member of a course"""
    ROLE = 'staff'
    def __init__(self, *args, **kwargs):
        super().__init__(self.ROLE, *args, **kwargs)
```

Reply:
@0x29a, yes, because the constructor of the `CourseStaffRole` does not expect the role as an explicit argument. Actually, we don't need to define the constructor of the `CourseLimitedStaffRole` at all.